compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -231,6 +232,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             _ => self.check_expr_kind(expr, expected),
         });
 
+        let is_uninhabited = || {
+            self.tcx.is_ty_uninhabited_from(
+                self.tcx.parent_module(self.body_id).to_def_id(),
```

Comment:
Wondering if this should be cached on FnCtxt?

Reply:
we can do a perf run to see if that's an issue (but I think you need to rebase first)